multinet/api/views/upload.py

```diff
@@ -0,0 +1,119 @@
+
+        try:
+            field_value = serializer.validated_data['field_value']
+            field_value_dict: Dict = signing.loads(field_value)
+            object_key = field_value_dict['object_key']
+        except signing.BadSignature:
+            return Response(
+                {'field_value': ['field_value is not a valid signed string.']},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        # Check for existing table name before creating upload and dispatching task
+        table_name = serializer.validated_data['table_name']
+        if Table.objects.filter(workspace=workspace, name=table_name).exists():
+            return Response(
+                {
+                    'table_name': [
+                        f'Table {table_name} in workspace {workspace.name} already exists.'
+                    ]
+                },
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        upload: Upload = Upload.objects.create(
+            workspace=workspace,
+            user=request.user,
+            blob=object_key,
+            data_type=Upload.DataType.CSV,
+        )
+
+        # Dispatch task
+        process_csv.delay(
+            upload.pk,
+            table_name=table_name,
+            edge=serializer.validated_data['edge'],
+            columns=serializer.validated_data['columns'],
+        )
+
+        return Response(UploadReturnSerializer(upload).data, status=status.HTTP_200_OK)
+
+    # @swagger_auto_schema(
+    #     request_body=UploadCreateSerializer(),
+    #     responses={200: UploadReturnSerializer()},
+    # )
+    # @method_decorator(
+    #     permission_required_or_403('owner', (Workspace, 'name', 'parent_lookup_workspace__name'))
+    # )
+    # @action(detail=False, url_path='d3_json', methods=['POST'])
+    # def upload_d3_json(self, request, parent_lookup_workspace__name: str):
+    #     """Create an upload of a D3 JSON file."""
+    #     workspace: Workspace = get_object_or_404(Workspace, name=parent_lookup_workspace__name)
+    #     serializer = UploadCreateSerializer(data=request.data)
+    #     serializer.is_valid(raise_exception=True)
+
+    #     try:
+    #         field_value = serializer.validated_data['field_value']
+    #         field_value_dict: Dict = signing.loads(field_value)
+    #         object_key = field_value_dict['object_key']
+    #     except signing.BadSignature:
+    #         return Response(
+    #             {'field_value': ['field_value is not a valid signed string.']},
+    #             status=status.HTTP_400_BAD_REQUEST,
+    #         )
+
+    #     upload: Upload = Upload.objects.create(
+    #         workspace=workspace,
+    #         user=request.user,
+    #         blob=object_key,
+    #         data_type=Upload.DataType.D3_JSON,
+    #     )
+
+    #     # TODO: Dispatch task
+
+    #     return Response(UploadReturnSerializer(upload).data, status=status.HTTP_200_OK)
```

Comment:
What's going on with this commented block?

Reply:
This is the frame around the d3_json uploader, but since it's not finished I didn't want to have the route included but doing nothing, so I commented it out. I should probably file issues to implement each of the remaining processors, this being one of them. Are you okay with keeping this comment block here until I come back to finish the implementation?

Reply:
Yeah that's fine. In general it's better to be explicit about missing functionality and keep WIP code in its on follow-on branch but in this case I don't want to disrupt your flow.

Reply:
I agree, my main reason for not minding this is that I'm likely going to work on it next, so it won't be here for too long. You're right though, I should have just separated it into its own branch.